zerver/tests/tests.py

```diff
@@ -587,6 +590,17 @@ def test_add_bot(self):
         bot = bots[0]
         self.assertEqual(bot['bot_owner'], 'hamlet@zulip.com')
 
+    def test_add_bot_with_user_avatar(self):
+        # type: () -> None
+        self.login("hamlet@zulip.com")
+        self.assert_num_bots_equal(0)
+        with open(os.path.join(TEST_AVATAR_DIR, 'img.png'), 'rb') as fp:
+            self.create_bot(file=fp)
+        self.assert_num_bots_equal(1)
+
+        profile = get_user_profile_by_email('hambot-bot@zulip.com')
+        self.assertEqual(profile.avatar_source, UserProfile.AVATAR_FROM_USER)
```

Comment:
Hmm, can't one just fetch the file at the user's avatar URL and compare the result (or its hash, or whatever) to `os.path.join(TEST_AVATAR_DIR, 'img.png')`?  I think it'd be fine to leave that for a follow-up issue though.